sqlmesh/core/model/meta.py

```diff
@@ -265,10 +267,10 @@ def normalized_cron(self) -> str:
             return "0 0 * * *"
         return ""
 
-    def croniter(self, value: TimeLike) -> croniter:
+    def croniter(self, value: TimeLike) -> CroniterCache:
         if self._croniter is None:
-            self._croniter = croniter(self.normalized_cron())
-        self._croniter.set_current(to_datetime(value))
+            self._croniter = CroniterCache(self.normalized_cron())
+        self._croniter.curr = value
```

Comment:
Why are we not passing this in constructor? We basically waste another `now` call.

Reply:
what do you mean?

Reply:
ah you mean value?

Reply:
yes, exactly